### DISCUSSION TITLE
How to generate a valid authorization signature?

### DISCUSSION THREAD

**Discussion by rodrigofaria** [minio/minio#16370] [Q&A] [answered]
I would like to use curl to download and upload files to my buckets in minio platform.

So after searching on the internet, I found this sample:
https://gist.github.com/JustinTimperio/ae695eef5fda1f1590a685a017bbb5ec

I try to use but I got the error:
_The request signature we calculated does not match the signature you provided. Check your key and signing method_

The only change that I made on the sample was to add `--insecure` in the curl, because I have no valid certificate from my server:

```
curl --insecure -o "${OUT_FILE}" \
    -H "Host: $URL" \
    -H "Date: ${DATE}" \
    -H "Content-Type: ${CONTENT_TYPE}" \
    -H "Authorization: AWS ${USERNAME}:${SIGNATURE}" \
    https://$URL${MINIO_PATH}
```

Also, I have tried with `http` and `https`.

My question is: how to generate a valid authorization signature?
Thanks!

**Comment by klauspost** [comment, marked as answer]
https://docs.aws.amazon.com/AmazonS3/latest/API/sig-v4-authenticating-requests.html

(No, it is not simple)
